ingestion_server/ingestion_server/ingest.py

```diff
@@ -288,9 +289,17 @@ def reload_upstream(
     shared_cols = _get_shared_cols(downstream_db, upstream_db, table)
     upstream_db.close()
 
-    with downstream_db.cursor() as downstream_cur:
-        # Step 2: Create FDW for the data transfer
+    with downstream_db, downstream_db.cursor() as downstream_cur:
+        # Step 2: Create the FDW extension if it does not exist
         log.info("(Re)initializing foreign data wrapper")
+        create_fdw = "CREATE EXTENSION IF NOT EXISTS postgres_fdw;"
+        try:
+            downstream_cur.execute(create_fdw)
+        except psycopg2.errors.UniqueViolation:
+            log.warning("Extension already exists, possible race condition.")
```

Comment:
Do you think this logic should exist in `queries.py` for consistency?

Reply:
That seems like a good idea, moved in 166d47c.